content/en/docs/kubernetes/operator/troubleshooting/automatic.md

```diff
@@ -0,0 +1,195 @@
+kubectl describe otelinst -n <namespace>
+```
+
+Where `<namespace>` is the namespace in which the `Instrumentation` resource is
+deployed.
+
+Your output should look like this:
+
+```yaml
+Name:         python-instrumentation
+Namespace:    application
+Labels:       app.kubernetes.io/managed-by=opentelemetry-operator
+Annotations:  instrumentation.opentelemetry.io/default-auto-instrumentation-apache-httpd-image:
+               ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-apache-httpd:1.0.3
+             instrumentation.opentelemetry.io/default-auto-instrumentation-dotnet-image:
+               ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-dotnet:0.7.0
+             instrumentation.opentelemetry.io/default-auto-instrumentation-go-image:
+               ghcr.io/open-telemetry/opentelemetry-go-instrumentation/autoinstrumentation-go:v0.2.1-alpha
+             instrumentation.opentelemetry.io/default-auto-instrumentation-java-image:
+               ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-java:1.26.0
+             instrumentation.opentelemetry.io/default-auto-instrumentation-nodejs-image:
+               ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-nodejs:0.40.0
+             instrumentation.opentelemetry.io/default-auto-instrumentation-python-image:
+               ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-python:0.39b0
+API Version:  opentelemetry.io/v1alpha1
+Kind:         Instrumentation
+Metadata:
+ Creation Timestamp:  2023-07-28T03:42:12Z
+ Generation:          1
+ Resource Version:    3385
+ UID:                 646661d5-a8fc-4b64-80b7-8587c9865f53
+Spec:
+...
+ Exporter:
+   Endpoint:  http://otel-collector-collector.opentelemetry.svc.cluster.local:4318
+...
+ Propagators:
+   tracecontext
+   baggage
+ Python:
+   Image:  ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-python:0.39b0
+   Resource Requirements:
+     Limits:
+       Cpu:     500m
+       Memory:  32Mi
+     Requests:
+       Cpu:     50m
+       Memory:  32Mi
+ Resource:
+ Sampler:
+Events:  <none>
+```
+
+### Check the OpenTelemetry Operator logs
+
+Check the OpenTelemetry Operator logs for errors by running this command:
+
+```shell
+kubectl logs -l app.kubernetes.io/name=opentelemetry-operator --container manager -n opentelemetry-operator-system --follow
+```
+
+The logs should not show any errors related to auto-instrumentation errors.
+
+### Check deployment order
+
+Make sure the deployment order is correct. The `Instrumentation` resource must
+be deployed before deploying the corresponding `Deployment` resources that are
+auto-instrumented.
+
+Consider the following auto-instrumentation annotation snippet:
+
+```yaml
+annotations:
+  instrumentation.opentelemetry.io/inject-python: 'true'
+```
+
+When the pod starts up, the annotation tells the Operator to look for an
+`Instrumentation` resource in the pod’s namespace, and to inject Python
```

Comment:
the instrumentation CR does not have to be in the pod namespace. There are a couple of possible values

```
The possible values for the annotation can be:

"true" - inject OpenTelemetryCollector resource from the namespace.
"sidecar-for-my-app" - name of OpenTelemetryCollector CR instance in the current namespace.
"my-other-namespace/my-instrumentation" - name and namespace of OpenTelemetryCollector CR instance in another namespace.
"false" - do not inject
```

Reply:
True, but I believe that in the example that I cite in [lines 87-99](https://github.com/avillela/opentelemetry.io/blob/dec09d83035244a9e956bff0ea77f5a6be442cae/content/en/docs/kubernetes/operator/troubleshooting/automatic.md?plain=1#L87-L99):

```yaml
annotations:
  instrumentation.opentelemetry.io/inject-python: 'true'
```

 it would still look for the `Instrumentation` resource in the pod's namespace, would it not? It only looks for it in a different namespace if the annotation includes the namespace, like with

```yaml
annotations:
  instrumentation.opentelemetry.io/inject-python: 'my-app-ns/my-instrumentation'
```

Is that not the case?

Reply:
Added more clarity [here](https://github.com/avillela/opentelemetry.io/blob/0b950a75b8d685359a2d3be6bc8b9699b7ee34d8/content/en/docs/kubernetes/operator/troubleshooting/automatic.md?plain=1#L279-L286)